test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ContractHTSSuite.java

```diff
@@ -478,6 +481,85 @@ private HapiApiSpec tokenTransferFromFeeCollector() {
 				);
 	}
 
+	private HapiApiSpec hbarTransferFromFeeCollector() {
+		return defaultHapiSpec("HbarTransferFromFeeCollector")
+				.given(
+						cryptoCreate(ACCOUNT).balance(10 * ONE_HUNDRED_HBARS).maxAutomaticTokenAssociations(10),
+						cryptoCreate(RECEIVER).maxAutomaticTokenAssociations(10),
+						cryptoCreate(TOKEN_TREASURY),
+						cryptoCreate(FEE_COLLECTOR),
+
+						fileCreate("bytecode").payingWith(ACCOUNT),
+						updateLargeFile(ACCOUNT, "bytecode", extractByteCode(ContractResources.HBAR_FEE_TRANSFER)),
+						fileCreate("nestedBytecode").payingWith(ACCOUNT),
+						updateLargeFile(ACCOUNT, "nestedBytecode",
+								extractByteCode(ContractResources.TRANSFERER)),
+						withOpContext(
+								(spec, opLog) -> {
+									allRunFor(spec, contractCreate(NESTED)
+											.payingWith(ACCOUNT)
+											.bytecode("nestedBytecode")
+											.gas(28_000));
+									allRunFor(
+											spec,
+											contractCreate(CONTRACT, HBAR_FEE_TRANSFER_CONSTRUCTOR, getNestedContractAddress(NESTED,
+													spec))
+													.payingWith(ACCOUNT)
+													.bytecode("bytecode")
+													.gas(28_000));
+									allRunFor(spec,
+											tokenCreate(A_TOKEN)
+													.tokenType(TokenType.FUNGIBLE_COMMON)
+													.initialSupply(TOTAL_SUPPLY)
+													.treasury(TOKEN_TREASURY)
+													.withCustom(fixedHbarFee(100L, CONTRACT)),
+
+											tokenAssociate(ACCOUNT, A_TOKEN),
+											tokenAssociate(RECEIVER, A_TOKEN),
+
+											cryptoTransfer(moving(TOTAL_SUPPLY, A_TOKEN)
+													.between(TOKEN_TREASURY, ACCOUNT)));
+								})
+				).when(
+						getAccountBalance(CONTRACT).hasTinyBars(0),
+						withOpContext(
+								(spec, opLog) -> {
+									final var token = asAddress(spec.registry().getTokenID(A_TOKEN));
+									final var sender = asAddress(spec.registry().getAccountID(ACCOUNT));
+									final var receiver = asAddress(spec.registry().getAccountID(RECEIVER));
+									final var feeCollector = asAddress(spec.registry().getAccountID(FEE_COLLECTOR));
+
+									/* --- HSCS-PREC-009 --- */
+									allRunFor(
+											spec,
+											contractCall(CONTRACT, HBAR_FEE_TRANSFER_DISTRIBUTE,
+													token,
+													sender,
+													receiver,
+													10L,
+													feeCollector
+											)
+													.payingWith(ACCOUNT)
+													.gas(48_000)
+													.via("distributeTx")
+													.alsoSigningWithFullPrefix(FEE_COLLECTOR)
+													.hasKnownStatus(SUCCESS));
+								})
+
+				).then(
+						childRecordsCheck("distributeTx", SUCCESS,
+								recordWith()
+										.status(SUCCESS)
+										.tokenTransfers(
+												changingFungibleBalances()
```

Comment:
In the child rec we also have `assessed_custom_fees` and `transferList`, can we also assert their correct values?